recursion/gnark-ffi/src/groth16.rs

```diff
@@ -201,6 +246,33 @@ impl Groth16Prover {
 
         deserialized
     }
+
+    /// Proves by sending a request to the Gnark server. If the server is not ready, it will wait
+    /// until it is before sending the request.
+    pub fn prove<C: Config>(&self, witness: Witness<C>) -> Groth16Proof {
+        self.wait_for_healthy_server().unwrap();
```

Comment:
Can we just wait in the call to new()? That way benchmarking is more accurate.

Reply:
sure although it might be nice to do other stuff while the download occurs

Reply:
moved it over to new()